nilearn/input_data/tests/test_nifti_masker.py

```diff
@@ -110,6 +111,39 @@ def test_mask_3d():
         assert_raises(TypeError, masker.fit)
 
 
+def test_mask_4d():
+    # Dummy mask
+    mask = np.zeros((10, 10, 10))
+    mask[3:7, 3:7, 3:7] = 1
+    mask_bool = mask.astype(bool)
+    mask_img = Nifti1Image(mask, np.eye(4))
+    n_mask_vox = mask_bool.sum()
+
+    # Dummy data
+    data = np.zeros((10, 10, 10, 3))
+    data[..., 0] = 1
+    data[..., 1] = 2
+    data[..., 2] = 3
+    data_img_4d = Nifti1Image(data, np.eye(4))
+    data_imgs = [Nifti1Image(data[..., 0], np.eye(4)),
+        Nifti1Image(data[..., 1], np.eye(4)),
+        Nifti1Image(data[..., 2], np.eye(4))]
```

Comment:
Yes. By the way, just curious, any particular reason you are not using data_img_4d in the transform call?